### DISCUSSION TITLE
Coloring of grids

### DISCUSSION THREAD

**Discussion by snowbollaanm** [zauberzeug/nicegui#1574] [Q&A] [answered]
for ui.grid() is there a way to color a specific column of a grid? 

I have made an element that serves as a timetable that is created through recursion and was wondering if there's a way to dynamically color a specific column of the ui.grid() element.

**Comment by rodja** [comment]
I don't think there is an elegant way to color grid cells/rows. The best I came up with:

```py
with ui.grid(columns=2).classes('gap-0'):
    ui.label('Name:')
    ui.label('Tom')

    ui.label('Age:').classes('bg-red')
    ui.label('42').classes('bg-red')

    ui.label('Height:')
    ui.label('1.80m')
```

**Reply by snowbollaanm** [reply]
Unfortunately that doesn't quite work as it isn't very readable and feels extremely cramped. 

Another way I was able to create it tho was this:
```py
from nicegui import ui

variable = 1
condition = 1
for i in range(10):
    with ui.column().classes('items-aligned') as scheduleColumn:
        if variable == condition:
            scheduleColumn.classes('bg-blue')
        ui.label('etc.')
        ui.label('etc.')

ui.run()
```
However, unfortunately that with the element I was attempting to create it does not line up correctly as it has: buttons and checkboxes so some columns do not have the same dimensions. Is there a way pass this where each element lines up equally?

**Reply by rodja** [reply]
Your example code does not show the problem you are describing which makes it harder to find a solution. Maybe we need to zoom out. Coloring of grids seems to be difficult. So maybe there is another way to achieve the visual appearance you are seeking. Could you share a minimal version of the code which builds up the structure? Also a graphic would help to illustrate what you want to achieve. Maybe some other css magic can be used...

**Comment by falkoschindler** [comment, marked as answer]
You could use a custom class and CSS to target "every n-th child":
```py
ui.add_head_html('''
<style>
.my-colored-grid > *:nth-child(3n+2) {
    background-color: #f0f0f0;
}
</style>
''')

with ui.grid(columns=3).classes('my-colored-grid'):
    ui.label('Name:')
    ui.label('Age:')
    ui.label('Height:')

    ui.label('Tom')
    ui.label('42')
    ui.label('1.80m')
```
